src/topology/algebra/ordered.lean

```diff
@@ -180,6 +181,27 @@ instance {α : Type*} [topological_space α] [partial_order α] [orderable_topol
   conv in (_ ∨ _) { rw or.comm }; refl⟩
 
 section partial_order
+
+section order_topology
+variable [partial_order α]
+
+/--
+(Order) topology on a partial order `α` generated by the subbase of open intervals
+`(a, ∞) = { x ∣ a < x }, (-∞ , b) = {x ∣ x < b}` for all `a, b` in `α`.
+-/
+def partial_order.topology : topological_space α :=
+generate_from {s : set α | ∃ (a : α), s = {b : α | a < b} ∨ s = {b : α | b < a}}
+
+local attribute [instance] partial_order.topology
+
+/--
+Order topology on a partial order `α` generated by the subbase of open intervals
+`(a, ∞) = { x ∣ a < x }, (-∞ , b) = {x ∣ x < b}` for all `a, b` in `α`.
+-/
+def partial_order.order_topology : orderable_topology α := ⟨rfl⟩
```

Comment:
The docstring is misleading, and it should be a lemma, not a def. This is a lemma that can be turned into an instance saying the construction `partial_order.topology` indeed produces an orderable topology.

Reply:
Changed it to `Shows that `partial_order.topology` indeed produces an orderable topology`.